src/modules/url-preview/generatePreview.js

```diff
@@ -7,16 +7,18 @@ const strategies = [
 ];
 
 export default async function generatePreview(url, cb) {
-	return new Promise((resolve) => {
+	return new Promise(async (resolve) => {
 		if (!/^https?:\/\//i.test(url)) {
 			cb();
 			return;
 		}
 
-		// add the async reduce function:
-		resolve(strategies.reduce(async (preview, strategy) => {
-			if (preview) return preview;
-			return await strategy(url);
-		}));
+		let preview;
+		for (let i = 0; i < strategies.length; i++) {
+			preview = await strategies[0](url); // eslint-disable-line babel/no-await-in-loop
+			if (preview) break;
+		}
+
+		resolve(preview || null);
```

Comment:
Reject when no preview there. Otherwise evreyone has to check if preview is `null` before using.